### PR TITLE
Clang tidy/geoipbackend

### PR DESCRIPTION
**Pull request by darnuria**
### Short description

An other clang-tidy PR.

- NULL -> nullptr
- Use std::array
- Use better overload for a `find` before a `string/char*` now a `char`.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code runned regtest `./start-test-stop 5300 geoip`
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: modules/geoipbackend/geoipbackend.cc [hunk 1]
```diff
@@ -649,13 +649,13 @@ string GeoIPBackend::format2str(string sformat, const Netmask& addr, GeoIPNetmas
   string::size_type cur, last;
   boost::optional<int> alt, prec;
   double lat, lon;
-  time_t t = time((time_t*)NULL);
+  time_t t = time(static_cast<time_t*>(nullptr));
```
Comment (rgacogne) [inline modules/geoipbackend/geoipbackend.cc:652]:
```suggestion
  time_t t = time(nullptr);
```

**Review by rgacogne — APPROVED** [review]
One tiny nit, looks good to me otherwise. Thanks!

File: modules/geoipbackend/geoipbackend.cc (hunk 1 above)
Comment (darnuria) [reply modules/geoipbackend/geoipbackend.cc:652]: Thanks: Overlooked this one, habit of C I presume.

**Review by rgacogne — APPROVED** [review]

**Comment by darnuria** [comment]
CI passed squashing the review commit :)

**Comment by rgacogne** [comment]
The CodeQL CI failure is unrelated.

**Comment by darnuria** [comment]
Also bump + rebase against `master`.

**Review by rgacogne — APPROVED** [review]

**Comment by darnuria** [comment]
Thanks!

**Pull request merged**